src/components/Calendars/ClientScheduleAppointment.tsx

```diff
@@ -0,0 +1,432 @@
+
+type ClientScheduleAppointmentProps = {
+    availabilities: Availability[];
+};
+
+export default function ClientScheduleAppointment({ availabilities }: ClientScheduleAppointmentProps) {
+    const MONTH_NAMES = [
+        'Janeiro',
+        'Fevereiro',
+        'Março',
+        'Abril',
+        'Maio',
+        'Junho',
+        'Julho',
+        'Agosto',
+        'Setembro',
+        'Outubro',
+        'Novembro',
+        'Dezembro',
+    ];
+
+    const DAYS = ['Dom', 'Seg', 'Ter', 'Qua', 'Qui', 'Sex', 'Sáb'];
+
+    const [month, setMonth] = useState(new Date().getMonth());
+    const [day, setDay] = useState(new Date().getDate());
+    const [year, setYear] = useState(new Date().getFullYear());
+    const [no_of_days, setNumDays] = useState<number[]>([]);
+    const [blankdays, setBlankDays] = useState<number[]>([]);
+    const [datepickerValue, setDatepickerValue] = useState('');
+    const [selectedHour, setSelectedHour] = useState<number | null>();
+
+    const [step, setStep] = useState(0);
+    const [availabilitiesMap, setAvailabilities] = useState<Map<string, number[]>>();
+    const [availableHours, setAvailableHours] = useState<number[]>();
+
+    // Calculates the number of days in the given month
+    const getNoOfDays = useCallback(
+        (month: number) => {
+            let daysInMonth = new Date(year, month + 1, 0).getDate();
+            let dayOfWeek = new Date(year, month).getDay();
+
+            let blankdaysArray: number[] = [];
+            for (var i = 1; i <= dayOfWeek; i++) {
+                blankdaysArray.push(i);
+            }
+
+            let daysArray: number[] = [];
+            for (var i = 1; i <= daysInMonth; i++) {
+                daysArray.push(i);
+            }
+
+            setBlankDays(blankdaysArray);
+            setNumDays(daysArray);
+        },
+        [year],
+    );
+
+    useEffect(() => {
+        getNoOfDays(month);
+    }, [month, getNoOfDays]);
+
+    useEffect(() => {
+        if (availabilities) {
+            const datesList = availabilities.map((d: Availability) => d.attributes.date);
+            const dateMap: Map<string, number[]> = new Map();
+
+            for (let d of datesList) {
+                const date = new Date(d).toLocaleDateString();
+                const hour = parseInt(d.substr(11, 2));
+                if (!dateMap.has(date)) {
+                    dateMap.set(date, []);
+                }
+                const hoursArray = dateMap.get(date);
+                if (hoursArray) hoursArray.push(hour);
+            }
+
+            setAvailabilities(dateMap);
+            setAvailableHours(dateMap.get(new Date().toLocaleDateString()));
+        }
+    }, [availabilities]);
+
+    // Checks if the day is the current day
+    const isToday = (date: number) => {
+        const today = new Date();
+        const d = new Date(year, month, date);
+
+        return today.toDateString() === d.toDateString() ? true : false;
+    };
```

Comment:
this function is recreated on every render, and also, instead you should use: https://date-fns.org/v2.30.0/docs/isToday